cats/cats-dynomite/src/test/groovy/com/netflix/spinnaker/cat/dynomite/cache/DynomiteCacheSpec.groovy

```diff
@@ -0,0 +1,264 @@
+    )
+  }
+
+  @Unroll
+  def 'attribute datatype handling #description'() {
+    setup:
+    def mergeData = createData('foo', [test: value], [:])
+    cache.merge('test', mergeData)
+
+    when:
+    def cacheData = cache.get('test', 'foo')
+
+    then:
+    cacheData != null
+    cacheData.attributes.test == expected
+
+    where:
+    value                      | expected                   | description
+    null                       | null                       | "null"
+    1                          | 1                          | "Integer"
+    2.0f                       | 2.0f                       | "Float"
+    "Bacon"                    | "Bacon"                    | "String"
+    true                       | true                       | "Boolean"
+    ['one', 'two']             | ['one', 'two']             | "Primitive list"
+    [key: 'value', key2: 10]   | [key: 'value', key2: 10]   | "Map"
+    new Bean('value', 10)      | [key: 'value', key2: 10]   | "Java object"
+    [key: 'value', key2: null] | [key: 'value']             | "Map with null"
+    new Bean('value', null)    | [key: 'value', key2: null] | "Java object with null"
+  }
+
+  @Unroll
+  def 'cache data will expire if ttl specified'() {
+    setup:
+    def mergeData = new DefaultCacheData('ttlTest', ttl, [test: 'test'], [:])
+    cache.merge('test', mergeData);
+
+    when:
+    def cacheData = cache.get('test', 'ttlTest')
+
+    then:
+    cacheData.id == mergeData.id
+
+    when:
+    Thread.sleep(Math.abs(ttl) * 1500)
+    cacheData = cache.get('test', 'ttlTest')
+
+    then:
+    cacheData?.id == (ttl > 0 ? null : mergeData.id)
+
+    where:
+    ttl || _
+    -1  || _
+    1   || _
+
+  }
+
+  def 'verify MSET chunking behavior (> MAX_MSET_SIZE)'() {
+    setup:
+    ((WriteableCache) cache).mergeAll('foo', [createData('bar'), createData('baz'), createData('bam')])
+
+    expect:
+    cache.getIdentifiers('foo').sort() == ['bam', 'bar', 'baz']
+  }
+
+  def 'should fail if maxMsetSize is not even'() {
+    when:
+    RedisCacheOptions.builder().maxMset(7).build()
+
+    then:
+    thrown(IllegalArgumentException)
+  }
+
+  def 'should not write an item if it is unchanged'() {
+    setup:
+    def data = createData('blerp', [a: 'b'])
+
+    when:
+    ((WriteableCache) cache).merge('foo', data)
+
+    then:
+    1 * cacheMetrics.merge('test', 'foo', 1, 1, 0, 0, 1, 2, 1, 0, 1, 0, 0)
+
+    when:
+    ((WriteableCache) cache).merge('foo', data)
+
+    then:
+    1 * cacheMetrics.merge('test', 'foo', 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0)
+  }
+
+  private static class Bean {
+    String key
+    Integer key2
+
+    Bean(String key, Integer key2) {
+      this.key = key
+      this.key2 = key2
+    }
+  }
+
+  private void initLocalDynoClusterClient() {
+    directClient = new Jedis('192.168.99.100', 22122)
+    directClient.flushAll()
+    directClient.close()
+
+    // This setup assumes that you're running a single-node Dynomite cluster via the Docker image.
```

Comment:
https://github.com/Netflix/dynomite/blob/dev/docker/HOWTO.md

Their official image didn't seem to start up correctly in Kitematic, so I built it locally. Running it, you just need to forward the 4 ports that it exposes (again, if Kitematic).